test/java/seedu/addressbook/util/TestUtil.java

```diff
@@ -75,4 +80,19 @@ public static void assertTextFilesEqual(Path path1, Path path2) throws IOExcepti
         List<String> list2 = Files.readAllLines(path2, Charset.defaultCharset());
         assertEquals(String.join("\n", list1), String.join("\n", list2));
     }
+
+    /**
+     * Executes the command, and asserts the result message is as expected.
+     */
+    public static void assertCommandResult(Command command, AddressBook addressBook,
+                                           AddressBook expectedAddressBook, String expectedMessage) {
```

Comment:
@zzzzwen 
> there is a TODO below which asks to implement equals method in CommandResult, then compare the expected CommandResult and actual

The point is that the code is not checking that the `CommandResult` returned by `ViewCommand` and `ViewAllCommand` has `getRelevantPersons() == Optional.empty()`.

> but in this case why there is false negative?

Let's say that we have `AddressBook` `a` and `b`, and a `ViewCommand` called `command`. We do `command.setData(a, ...)` but then pass `b` into `assertCommandResult()`, e.g. `assertCommandResult(command, b, TestUtil.clone(b), ...)`.

Now, let's say that `command` has a bug that ends up modifying the address book `a`. However, with the current implementation of `assertCommandResult()`, the test will still pass since `b` was not modified, even though `command` has a bug.